torch/_guards.py

```diff
@@ -456,24 +456,36 @@ def detect_fake_mode(inputs: Any = None):
     from torch._subclasses.fake_tensor import FakeTensor, FakeTensorMode
     from torch.utils._pytree import tree_flatten
 
+    fake_modes = []
+
     context = TracingContext.get()
     if context is not None:
         fake_mode = context.fake_mode
         if fake_mode is not None:
-            return fake_mode
+            fake_modes.append((fake_mode, lambda: "tracing context"))
 
     from torch.utils._python_dispatch import _get_current_dispatch_mode_stack
 
-    for m in reversed(_get_current_dispatch_mode_stack()):
+    for i, m in enumerate(reversed(_get_current_dispatch_mode_stack())):
```

Comment:
nit: no reason to reverse the mode stack if we're asserting a single fake mode anyway

Reply:
I thought about this, but I figured I still wanted to report it "in order" anyway